src/systemctl/systemctl.c

```diff
@@ -559,7 +559,7 @@ static int output_units_list(const UnitInfo *unit_infos, unsigned c) {
                         off = ansi_normal();
                 }
 
-                if (arg_all)
+                if (arg_all || arg_states)
                         printf("%s%u loaded units listed.%s\n"
                                "To show all installed unit files use 'systemctl list-unit-files'.\n",
                                on, n_shown, off);
```

Comment:
Hmm, if I say `systemctl --failed`, it makes no sense to use `systemctl list-unit-files`. So I think this needs to be changed to be smarter, and not give a hint in that case. And it'd be even better to print the status in **green** if `--failed` is used — after all it's a good thing to see nothing here.

Reply:
So you'd like me to add a condition for `n_shown == 0` in the `--failed` case, to show a "0 units" message as green instead of bold.

I thought the reminder was useful for `systemctl --state=inactive`.  I'll have to think a bit more because `arg_states` is a `strv`.  E.g. if I use `systemctl --state=active --state=failed`, I don't want the green colouring when there are no units.

Reply:
I#d suppress the hint in the slightest doubt it makes sense. No hint is better than a confusing hint.